lib/mixpanel/uri.rb

```diff
@@ -15,7 +18,13 @@ def self.mixpanel(resource, params)
     end
 
     def self.encode(params)
-      params.map { |key, val| "#{key}=#{CGI.escape(val.to_s)}" }.sort.join('&')
+      params.map do |key, val|
+        if val.is_a?(Hash) || val.is_a?(Array)
```

Comment:
What do you think about something like:

    def self.encode(params)
      args.map { |key, value| "#{key}=#{format_value(value)}" }
    end

    def self.generate_signature(args, api_secret)
      Digest::MD5.hexdigest(
        args.map do |key, value|
          "#{key}=#{CGI.escape(format_value(value)}" 
        end.sort.join + api_secret
      )
    end

    def format_value(value)
      return value.to_json if [Array, Hash].include?(value)
      value.to_s
    end